### DISCUSSION TITLE
Тип плагина в Joomla 4

### DISCUSSION THREAD

**Discussion by dzandut** [JPathRu/localisation#56] [General]
Хочу узнать мнение команды.

В переводе для J3 часть типов плагина переведена, а более новая часть нет. (например поля).
В итоге в списке плагинов визуально это смотрится не очень хорошо.

Лично мне очень нравилось когда все было на Русском, но уже на тот момент было 3 особенности этого решения.
1. При использовании фильтра использовалось английское название (_нет смысла переделывать из пунктов ниже_)
2. Большинство сторонних плагинов которые даже имели локализацию зачастую не имели перевода для типов.
3. Часть плагинов не была локализована совсем

Пример:
![image](https://user-images.githubusercontent.com/33223568/111875429-ff3b2580-8999-11eb-8ed2-88d9aae2036b.png)

Скажу свои за и против относительно перевода на Русский типов плагина.
Плюсы:
- Гораздо легче воспринимается
- Легче ориентироваться
- Лучше понятно назначение плагина (_так как большая часть плагинов связана с другими компонентами и благодаря идентичному названию легче понять к чему относится этот плагин_)
- Эстетически приятней (_субъективно_)

Минусы:
- Получается микс из русских и английских названии
- Смущает отличие в выпадающем списке от результата 

Что вы думаете. Стоит ли в переводе для J4 делать перевод типов плагина?
Если нет то стоит ли переводит эту часть заголовка в плагине?
![image](https://user-images.githubusercontent.com/33223568/111875448-111cc880-899a-11eb-880b-928a37293b83.png)

Спасибо за участие

**Comment by progreccor** [comment]
мое личное мнение - не делать перевод тип плагина.
проще будет всем

поясню более развернуто - для большинства непрофессионалов тип плагина не имеет никакого смысла и значения - для них это абракадабра. Так пусть она такой и останется.

для разработчиков имеет плюс - тип плагина = имени папки. так проще ориентироваться.

поэтому я за то, чтобы не переводить тип плагина

**Reply by dzandut** [reply]
> для разработчиков имеет плюс - тип плагина = имени папки. так проще ориентироваться.
С этим согласен

**Reply by Kostelano** [reply]
@dzandut В Joomla 3 полностью все плагины русифицированы. Почему в твоей версии частично? Странно :)

Что касается типов/названия плагинов, то я ЗА перевод на русский, поскольку в этом и есть смысл локализации на тот или иной язык + совсем будет абракадабра, если в списке компонентов (например) увидеть название "Рабочий процесс" (условно), а в списке плагинов оригинальное **Workflow**. Нужно как-то "в одну сторону" с таким моментом.

**Comment by Umasterov** [comment]
Я за перевод. Вроде тут для этого все и собрались

**Reply by progreccor** [reply]
> 
> 
> Я за перевод. Вроде тут для этого все и собрались

на ясность использования наплевать?
в css например никто не переводит background и другие правила...
мне кажется надо соблюдать разумные рамки

**Reply by Umasterov** [reply]
> на ясность использования наплевать?

Ясность от этого только улучшится

> в css например никто не переводит background и другие правила...

CSS вообще никто не переводит

> мне кажется надо соблюдать разумные рамки

Тут согласен

Кому удобно работать с англоязычным интерфейсом -- у них всегда остаётся возможность поставить английский язык. Нужно дать такую же возможность тем, кому удобнее работать с рускоязычным интерфейсом

**Comment by progreccor** [comment]
есть и еще один аргумент  - посмотрите как выглядят плагины, установленные от сторонних расширений:

[
![Скриншот 20-03-2021 203756](https://user-images.githubusercontent.com/3103677/111880378-43d7b880-89bc-11eb-92a1-8bd19ba9f3d6.png)
](url)

вам не кажется, что это приводит к дополнительной путанице?

Иначе мы придем к тому чтобы переводить images в папке изображений...

если именно мое мнение - то переводить типы плагинов в предыдущей версии joomla было ошибкой

**Reply by Umasterov** [reply]
Будет хороший пример для переводчиков плагинов и стимул переводить их названия

**Reply by progreccor** [reply]
> 
> 
> Будет хороший пример для переводчиков плагинов и стимул переводить их названия

в некоторых расширениях названия захардкодены и никто их переводить не будет. И вот эта мешанина будет всегда.
Она и сейчас меня бесит и дальше ничего не изменится.

**Reply by Umasterov** [reply]
> в некоторых расширениях названия захардкодены и никто их переводить не будет. И вот эта мешанина будет всегда.
> Она и сейчас меня бесит и дальше ничего не изменится.

Не стоит ориентироваться на плохие практики. А то так и вовсе можно от перевода отказаться

**Comment by dzandut** [comment]
Я затрудняюсь ответить как лучше так как минусы и плюсы есть у обоих вариантов.
Если переводить то языковая мешанина получается, а если не переводить соответствия компонентам явного не будет.

**Comment by b2z** [comment]
Я за перевод

**Comment by Kostelano** [comment]
Не понимаю о какой мешанине речь. Есть русский язык - на него все переведено.

Если устанавливаешь чужое расширение из Нидерландов, само собой, что там будет инглиш. Если разработчик не позаботился о том, чтобы в XML-файл плагина вбить константу, а не название на инглише, ну что ж тут поделаешь. Но отказываться от перевода группы плагинов только из-за этого, как по мне, неверно.

**Reply by b2z** [reply]
Плюс
